src/TypeSystem/Type/UnionType/UnionType.php

```diff
@@ -22,17 +22,24 @@
 
 namespace PackageFactory\ComponentEngine\TypeSystem\Type\UnionType;
 
+use PackageFactory\ComponentEngine\TypeSystem\Type\NullType\NullType;
 use PackageFactory\ComponentEngine\TypeSystem\TypeInterface;
 
-final class UnionType implements TypeInterface
+/**
+ * @implements \IteratorAggregate<int, TypeInterface>
+ */
+final class UnionType implements TypeInterface, \IteratorAggregate, \Countable
 {
     /**
      * @var TypeInterface[]
      */
-    private array $members;
+    private readonly array $members;
 
     private function __construct(TypeInterface ...$members)
     {
+        if (count($members) < 1) {
```

Comment:
Wow thanks, great idea with enforcing it structurally `TypeInterface $firstMember, TypeInterface ...$members` didnt think about this ^^

i kept the 

```
assert(count($members) > 1, 'UnionType must hold at least two different members');
```

in the constructor, since i like having the rules of objects enforced in the constructor (in case someone creates a new static factory or a `with*` method, which invokes the private constructor directly ...)